### PR TITLE
Only process result and lifespan based on duration

### PR DESCRIPTION
**Pull request by Mac-Ste**
Added process only test
Added separation by action result report
Added lifespan according to test duration

### CODE REVIEW

File: src/test/kotlin/JiraPerformanceComparisonIT.kt [hunk 1]
```diff
@@ -44,22 +50,48 @@ class JiraPerformanceComparisonIT {
...
             results = results,
-            workspace = workspace.isolateTest("Compare")
+            workspace = currentTaskWorkspace.isolateTest("Compare")
         )
         dumpMegaSlowWaterfalls(results)
     }
 
+    @Test
+    fun shouldOnlyProcessGatheredData() {
+        val task = "2020-03-06T14-00-16.448"
```
Comment (dagguh) [inline src/test/kotlin/JiraPerformanceComparisonIT.kt]: It won't work for me, because I don't have this data. It's fine to have this test, but it shouldn't be bound to the `comparePerformance` task. I'd extract it to a separate `IT`.

**Review by dagguh — CHANGES_REQUESTED** [review]

File: src/test/kotlin/JiraPerformanceComparisonIT.kt (hunk 1 above)
Comment (Mac-Ste) [reply src/test/kotlin/JiraPerformanceComparisonIT.kt]: Updated the code, moved all to separate test class and created separate gradle task for it

File: src/test/kotlin/JiraPerformanceComparisonProcessOnlyIT.kt [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import com.atlassian.performance.tools.report.api.FullTimeline
+import com.atlassian.performance.tools.report.api.Timeline
+import com.atlassian.performance.tools.report.api.result.EdibleResult
+import com.atlassian.performance.tools.report.api.result.RawCohortResult
+import com.atlassian.performance.tools.workspace.api.RootWorkspace
+import org.junit.Test
+import java.io.File
+import java.nio.file.Paths
+
+class JiraPerformanceComparisonProcessOnlyIT {
```
Comment (dagguh) [inline src/test/kotlin/JiraPerformanceComparisonProcessOnlyIT.kt]: It's a bit wordy. We can avoid these hard to remember names by simplifying. Most of the name tries to explain what it _does not do_. Let's focus on what it actually does, e.g. `ExistingResultsIT.shouldReport()`

File: src/test/kotlin/JiraPerformanceComparisonIT.kt [hunk 3]
```diff
@@ -55,32 +50,6 @@ class JiraPerformanceComparisonIT {
         dumpMegaSlowWaterfalls(results)
     }
 
-    @Test
-    fun shouldOnlyProcessGatheredData() {
```
Comment (dagguh) [inline src/test/kotlin/JiraPerformanceComparisonIT.kt]: Let's squash the PR

**Review by dagguh — CHANGES_REQUESTED** [review]

File: src/test/kotlin/JiraPerformanceComparisonIT.kt (hunk 3 above)
Comment (dagguh) [reply src/test/kotlin/JiraPerformanceComparisonIT.kt]: Oops, I meant that we should squash the two "process only" commits (first in JPCIT, second in a separate IT). But the independent lifespan change got squashed in too. Unsquashing and restitching (`git reset`, `git add -p`) is worth the effort in general, but with our current deadlines it's ok to leave it oversquashed.

**Review by dagguh — APPROVED** [review]
